nixos/modules/virtualisation/xen/xl.nix

```diff
@@ -0,0 +1,55 @@
+# Xen Project Hypervisor Domain 0 libxenlight configuration
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib) mkIf mkOption;
+
+  cfg = config.virtualisation.xen;
+
+  settingsFormat = pkgs.formats.xenLight { type = "conf"; };
+in
+{
+  ## Interface ##
+
+  options.virtualisation.xen = {
+    settings = mkOption {
+      inherit (settingsFormat) type;
+      default = { };
+      example = {
+        autoballoon = "off";
+        bootloader_restrict = false;
+        lockfile = "/run/lock/xen/xl";
+        max_grant_version = 256;
+        "vif.default.bridge" = "xenbr0";
+        "vm.hvm.cpumask" = [
+          "2"
+          "3-8,^5"
+        ];
+      };
+      description = ''
+        The contents of the `/etc/xen/xl.conf` file.
+        See {manpage}`xl.conf(5)` for available configuration options.
+      '';
+    };
+  };
+
+  ## Implementation ##
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.dom0Resources.memory != 0 -> !cfg.settings.autoballoon == "on";
```

Comment:
The type and the default value of this `autoballoon` is not defined, so if I have `dom0Resources` I will receive `error: attribute 'autoballoon' missing`, and if I set `autoballoon = "off";` I will get `error: expected a Boolean but found a string: "off"`